lib/src/call/quality_issue.dart

```diff
@@ -0,0 +1,343 @@
+/// the event will not be invoked with the level `VIQualityIssueLevel.None` in cases:
+///
+/// * The (conference) call ended
+/// * The endpoint left the conference call -
+/// [VIEndpoint.onEndpointRemoved] is invoked
+///
+/// The issue is not detected for the following cases:
+///
+/// * The endpoint put the call on hold via [VICall.hold]
+/// * The endpoint stopped sending audio during a call via [VICall.sendAudio]
+///
+/// Possible reasons:
+///
+/// * Poor internet connection on the client or the endpoint
+/// * Connection lost on the endpoint
+class NoAudioReceive extends QualityIssue {
+  /// Id remote audio stream the issue occured on.
+  final String audiostreamId;
+
+  /// Id endpoint the issue belongs to.
+  final String endpointId;
+
+  NoAudioReceive._fromMap(Map<dynamic, dynamic> map)
+      : this.audiostreamId = map['audiostreamId'],
+        this.endpointId = map['endpointId'],
+        super._fromMap(map);
+}
+
+/// Represents class reporting that no video is received on the
+/// remote video stream.
+///
+/// Issue level can be only `VIQualityIssueLevel.Critical`
+/// if the issue is detected or `VIQualityIssueLevel.None`
+/// if the issue is not detected or resolved.
+///
+/// If no video receive is detected on several remote video streams,the
+/// event will be invoked for each of the remote video streams with the issue.
+///
+/// If the issue level is `VIQualityIssueLevel.Critical`
+/// the event will not be invoked with the level `VIQualityIssueLevel.None` in cases:
+///
+/// * The (conference) call ended
+/// * The remote video stream was removed -
+/// [VIEndpoint.onRemoteVideoStreamRemoved] is invoked
+/// * The endpoint left the conference call -
+/// [VIEndpoint.onEndpointRemoved] is invoked
+///
+/// The issue is not detected for the following cases:
+///
+/// * The endpoint put the call on hold via [VICall.hold]
+/// * The endpoint stopped sending audio during a call via [VICall.sendVideo]
+/// * Video receiving was stopped on the remote video stream via
+/// [VIEndpoint.stopReceiving]
+///
+/// Possible reasons:
+///
+/// * Poor internet connection on the client or the endpoint
+/// * Connection lost on the endpoint
+/// * The endpoint's application has been moved to the background state on an
+/// iOS device (camera usage is prohibited while in the background on iOS)
+class NoVideoReceive extends QualityIssue {
+  /// Id remote video stream the issue occured on.
+  final String videostreamId;
+
+  /// Id endpoint the issue belongs to.
+  final String endpointId;
+
+  NoVideoReceive._fromMap(Map<dynamic, dynamic> map)
+      : this.videostreamId = map['videostreamId'],
+        this.endpointId = map['endpointId'],
+        super._fromMap(map);
+}
+
+/// Represents a quality issue.
+class _VIQualityIssue {
+  StreamController<QualityIssue> _qualityStreamController =
+      StreamController.broadcast();
+
+  _VIQualityIssue._() {
+    subscribeToIssues();
+  }
+
+  void subscribeToIssues() {
```

Comment:
should it be private?

Reply:
Why? we have private class _VIQualityIssue, and we do not can create instance this class for using this method

Reply:
I do not see the reasons to keep it public, this class is private and all its fields and functions are also private.

Indicating it as public leads to misunderstanding about the intents of this method